server_modules/file_system.js

```diff
@@ -0,0 +1,133 @@
+var fs = require('fs');
+var Q = require('q');
+var mkdirp = require('mkdirp');
+var path = require('path')
+var utils = require('./utils');
+var args = require('./args');
+
+function getDataFilePath(file) {
+    return exports.resolve(path.dirname(process.argv[1]/*TODO change this to user the args module*/), args.data, file);
+}
+
+function parseFile(data) {
+	return Q.promise(function(resolve,reject) {
+        try {
+            var res = JSON.parse(data);
+            resolve(res);
+        } catch(e) {
+            reject(e);
+        }
+    });
+}
+
+exports.resolve = function(file) {
+	if(file[0] === '/') {
+		file = utils.root + file;
+	}
+	return path.resolve(file);
+};
+
+exports.readFile = function(file) {
+	if(file[0] === '/') {

```

Comment:
This now seems asymmetric compared to `writeFile`, so one of them is probably wrong now.
I'm not sure exactly where `readFile` and `writeFile` are used exactly, but it seems a few use-cases are being mixed now: serving files from the `/fs` tree, vs serving generic files from the filesystem.